### PR TITLE
Add Nepal country support and unit test

### PR DESCRIPTION
**Pull request by Summonshr**

### CODE REVIEW

File: src/Countries/Nepal.php [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+    public function countryCode(): string
+    {
+        return 'np';
+    }
+
+    protected function allHolidays(int $year): array
+    {
+        return array_merge([
+            'New Year' => '01-01',
```
Comment (ntn0de) [inline src/Countries/Nepal.php:22]: I don't think "English New Year" counts as a public holiday

File: src/Countries/Nepal.php [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+            'International Labour Day' => '05-01',
+            'Christmas Day' => '12-25',
+        ], []);
+    }
+
+    /** @return array<string, CarbonImmutable> */
+    protected function variableHolidays(int $year): array
+    {
+        return [
+            // Return specific holidays for this year
```
Comment (Nielsvanpach) [inline src/Countries/Nepal.php:32]: Todo

File: tests/Countries/NepalTest.php [hunk 3]
```diff
@@ -0,0 +1,18 @@
+<?php
+
+namespace Spatie\Holidays\Tests\Countries;
+
+use Carbon\CarbonImmutable;
+use Spatie\Holidays\Holidays;
+
+it('can calculate nepal holidays', function () {
+    CarbonImmutable::setTestNowAndTimezone('2024-01-01');
```
Comment (Nielsvanpach) [inline tests/Countries/NepalTest.php:9]:
```suggestion
    CarbonImmutable::setTestNow('2024-01-01');
```

Comment (Nielsvanpach) [reply tests/Countries/NepalTest.php:9]: Could you also run the tests locally and verify the generated results in the snapshot?

**Comment by Nielsvanpach** [comment]
Thanks, but I'm closing this PR as it doesn't feel complete yet. Feel free to reopen once fixed!

**Pull request closed without merging**